hibernate-core/src/main/java/org/hibernate/internal/util/compare/EqualsHelper.java

```diff
@@ -16,6 +16,16 @@
  */
 public final class EqualsHelper {
 
+	/**
+	 * Compares two objects for equality, handles null.
+	 *
+	 * @param x the first object, possibly null
+	 * @param y the second object, possibly null
+	 * @return true if the two objects are equal either by identity
+	 *         or using their equals method
+	 * @deprecated use {@link java.util.Objects#equals(Object, Object)}
+	 */
+	@Deprecated
 	@SuppressWarnings("SimplifiableIfStatement")
 	public static boolean equals(final Object x, final Object y) {
 		if ( x == y ) {
```

Comment:
IMO this should be changed to call `Objects#equals`

Reply:
@sebersole considering it's internal and shouldn't be used anymore in our code, shouldn't we simply remove it?

Reply:
@marschall could you simply remove this method from the code? The idea is to not use it anymore in our code and people shouldn't use our internel helpers either.

Thanks!

Reply:
Sure, no problem.